homeassistant/components/websocket_api/connection.py

```diff
@@ -268,6 +271,10 @@ def async_handle_exception(self, msg: dict[str, Any], err: Exception) -> None:
             err_message = "Timeout"
         elif isinstance(err, HomeAssistantError):
             err_message = str(err)
+            code = const.ERR_UNKNOWN_ERROR
```

Comment:
It's misleading to call HomeAssistantError an unknown error. Those are never unknown but are raised for specific cases. Maybe have a code for HomeAssistantError?

Reply:
True, but with changing this we will need to update a lot of tests. I suggest we change that in a follow-up PR

Reply:
Ok, good.

Reply:
I'll link a new PR adressing this.

Reply:
https://github.com/home-assistant/core/pull/104700